Style errors with ruff v0.9.0

[ruff v0.9.0](https://github.com/astral-sh/ruff/releases/tag/0.9.0) was released a few minutes ago. Refer to the [blog post](https://astral.sh/blog/ruff-v0.9.0) for details.

There are some new errors now:
```
ruff check pygmt doc/conf.py examples
pygmt/encodings.py:1:1: A005 Module `encodings` shadows a Python standard-library module
pygmt/helpers/tempfile.py:1:1: A005 Module `tempfile` shadows a Python standard-library module
pygmt/io.py:1:1: A005 Module `io` shadows a Python standard-library module
pygmt/src/select.py:1:1: A005 Module `select` shadows a Python standard-library module
Found 4 errors.
make: *** [Makefile:67: check] Error 1
```

We need to either fix them or ignore them

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -210,12 +210,10 @@ html_theme_options = {}
 repository = "GenericMappingTools/pygmt"
 repository_url = "https://github.com/GenericMappingTools/pygmt"
 if __commit__:
-    commit_link = (
-        f'<a href="{repository_url}/commit/{ __commit__ }">{ __commit__[:8] }</a>'
-    )
+    commit_link = f'<a href="{repository_url}/commit/{__commit__}">{__commit__[:8]}</a>'
 else:
     commit_link = (
-        f'<a href="{repository_url}/releases/tag/{ __version__ }">{ __version__ }</a>'
+        f'<a href="{repository_url}/releases/tag/{__version__}">{__version__}</a>'
     )
 html_context = {
     "menu_links": [

--- a/environment.yml
+++ b/environment.yml
@@ -27,7 +27,7 @@ dependencies:
     # Dev dependencies (style checks)
     - codespell
     - pre-commit
-    - ruff>=0.8.2
+    - ruff>=0.9.0
     # Dev dependencies (unit testing)
     - matplotlib-base
     - pytest>=6.0

--- a/examples/tutorials/advanced/cartesian_histograms.py
+++ b/examples/tutorials/advanced/cartesian_histograms.py
@@ -348,7 +348,7 @@ fig.histogram(
     # of the bin width
     # Offset ("+o") the bars to align each bar with the left limit of the corresponding
     # bin
-    barwidth=f"{binwidth/2}+o-{binwidth/4}",
+    barwidth=f"{binwidth / 2}+o-{binwidth / 4}",
     label="data01",
 )
 
@@ -359,7 +359,7 @@ fig.histogram(
     fill="orange",
     pen="1p,darkgray,solid",
     histtype=0,
-    barwidth=f"{binwidth/2}+o{binwidth/4}",
+    barwidth=f"{binwidth / 2}+o{binwidth / 4}",
     label="data02",
 )
 

--- a/pygmt/_show_versions.py
+++ b/pygmt/_show_versions.py
@@ -16,7 +16,7 @@ from packaging.version import Version
 from pygmt.clib import Session, __gmt_version__
 
 # Get semantic version through setuptools-scm
-__version__ = f'v{version("pygmt")}'  # e.g. v0.1.2.dev3+g0ab3cd78
+__version__ = f"v{version('pygmt')}"  # e.g. v0.1.2.dev3+g0ab3cd78
 __commit__ = __version__.split("+g")[-1] if "+g" in __version__ else ""  # 0ab3cd78
 
 

--- a/pygmt/encodings.py
+++ b/pygmt/encodings.py
@@ -1,3 +1,4 @@
+# noqa: A005
 """
 Character encodings supported by GMT.
 

--- a/pygmt/helpers/tempfile.py
+++ b/pygmt/helpers/tempfile.py
@@ -1,3 +1,4 @@
+# noqa: A005
 """
 Utilities for dealing with temporary file management.
 """

--- a/pygmt/helpers/utils.py
+++ b/pygmt/helpers/utils.py
@@ -574,9 +574,8 @@ def launch_external_viewer(fname: str, waiting: float = 0) -> None:
     }
 
     match sys.platform:
-        case name if (
-            (name == "linux" or name.startswith("freebsd"))
-            and (xdgopen := shutil.which("xdg-open"))
+        case name if (name == "linux" or name.startswith("freebsd")) and (
+            xdgopen := shutil.which("xdg-open")
         ):  # Linux/FreeBSD
             subprocess.run([xdgopen, fname], check=False, **run_args)  # type:ignore[call-overload]
         case "darwin":  # macOS

--- a/pygmt/io.py
+++ b/pygmt/io.py
@@ -1,3 +1,4 @@
+# noqa: A005
 """
 PyGMT input/output (I/O) utilities.
 """

--- a/pygmt/src/grd2xyz.py
+++ b/pygmt/src/grd2xyz.py
@@ -145,8 +145,7 @@ def grd2xyz(
 
     if kwargs.get("o") is not None and output_type == "pandas":
         msg = (
-            "If 'outcols' is specified, 'output_type' must be either 'numpy' "
-            "or 'file'."
+            "If 'outcols' is specified, 'output_type' must be either 'numpy' or 'file'."
         )
         raise GMTInvalidInput(msg)
     # Set the default column names for the pandas DataFrame header.

--- a/pygmt/src/select.py
+++ b/pygmt/src/select.py
@@ -1,3 +1,4 @@
+# noqa: A005
 """
 select - Select data table subsets based on multiple spatial criteria.
 """

--- a/pygmt/tests/test_clib_virtualfile_from_stringio.py
+++ b/pygmt/tests/test_clib_virtualfile_from_stringio.py
@@ -43,14 +43,9 @@ def test_virtualfile_from_stringio():
     Test the virtualfile_from_stringio method.
     """
     data = io.StringIO(
-        "# Comment\n"
-        "H 24p Legend\n"
-        "N 2\n"
-        "S 0.1i c 0.15i p300/12 0.25p 0.3i My circle\n"
+        "# Comment\nH 24p Legend\nN 2\nS 0.1i c 0.15i p300/12 0.25p 0.3i My circle\n"
     )
-    expected = (
-        ">\n" "H 24p Legend\n" "N 2\n" "S 0.1i c 0.15i p300/12 0.25p 0.3i My circle\n"
-    )
+    expected = ">\nH 24p Legend\nN 2\nS 0.1i c 0.15i p300/12 0.25p 0.3i My circle\n"
     assert _stringio_to_dataset(data) == expected
 
 
@@ -66,13 +61,7 @@ def test_one_segment():
         "6 7 8   9  FGHIJK LMN OPQ\n"
         "RSTUVWXYZ\n"
     )
-    expected = (
-        "> Segment 1\n"
-        "1 2 3 ABC\n"
-        "4 5 DE\n"
-        "6 7 8   9  FGHIJK LMN OPQ\n"
-        "RSTUVWXYZ\n"
-    )
+    expected = "> Segment 1\n1 2 3 ABC\n4 5 DE\n6 7 8   9  FGHIJK LMN OPQ\nRSTUVWXYZ\n"
     assert _stringio_to_dataset(data) == expected
 
 

--- a/pygmt/tests/test_grdview.py
+++ b/pygmt/tests/test_grdview.py
@@ -161,7 +161,7 @@ def test_grdview_with_perspective_and_zaxis_frame(xrgrid, region):
     a Transverse Mercator (T) projection.
     """
     fig = Figure()
-    projection = f"T{(region[0]+region[1])/2}/{abs((region[2]+region[3])/2)}"
+    projection = f"T{(region[0] + region[1]) / 2}/{abs((region[2] + region[3]) / 2)}"
     fig.grdview(
         grid=xrgrid,
         projection=projection,

--- a/pygmt/tests/test_info.py
+++ b/pygmt/tests/test_info.py
@@ -23,10 +23,7 @@ def test_info():
     """
     output = info(data=POINTS_DATA)
     expected_output = (
-        f"{POINTS_DATA}: N = 20 "
-        "<11.5309/61.7074> "
-        "<-2.9289/7.8648> "
-        "<0.1412/0.9338>\n"
+        f"{POINTS_DATA}: N = 20 <11.5309/61.7074> <-2.9289/7.8648> <0.1412/0.9338>\n"
     )
     assert output == expected_output
 
@@ -57,10 +54,7 @@ def test_info_path(table):
     """
     output = info(data=table)
     expected_output = (
-        f"{POINTS_DATA}: N = 20 "
-        "<11.5309/61.7074> "
-        "<-2.9289/7.8648> "
-        "<0.1412/0.9338>\n"
+        f"{POINTS_DATA}: N = 20 <11.5309/61.7074> <-2.9289/7.8648> <0.1412/0.9338>\n"
     )
     assert output == expected_output
 

